src/chainstate/stacks/boot/pox-2.clar

```diff
@@ -327,18 +327,48 @@
                                  cycle-entry-total-ustx) })
             (ok (merge data { cycle: (+ u1 cycle)} )))))
 
+(define-private (get-event-info (stacker principal))
```

Comment:
I think you're definitely going to want to run this step outside of the contract invocation.  This takes two `read_counts`, and in some cases it's used within a `fold`.